src/Directory.Build.props

```diff
@@ -1,29 +1,47 @@
 <Project>
-  <PropertyGroup>
+
+  <PropertyGroup Label="Build configuration">
+    <LangVersion>7.3</LangVersion>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)..\obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
     <OutputPath>$(MSBuildThisFileDirectory)..\bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
     <PackageOutputPath>$(MSBuildThisFileDirectory)..\bin\Packages\$(Configuration)\</PackageOutputPath>
+    <IsTestProject>$(MSBuildProjectName.Contains('Test'))</IsTestProject>
+    <IsPackable Condition=" $(IsTestProject) ">false</IsPackable>
+    <GenerateDocumentationFile Condition=" !$(IsTestProject) ">true</GenerateDocumentationFile>
+    <CodeAnalysisRuleSet Condition=" !$(IsTestProject) ">$(MSBuildThisFileDirectory)/CodeGeneration.Roslyn.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet Condition=" $(IsTestProject) ">$(MSBuildThisFileDirectory)/CodeGeneration.Roslyn.Tests.ruleset</CodeAnalysisRuleSet>
+    <WarningsAsErrors>true</WarningsAsErrors>
+  </PropertyGroup>
+
+  <PropertyGroup Label="NuGet metadata">
     <PackageProjectUrl>https://github.com/aarnott/CodeGeneration.Roslyn</PackageProjectUrl>
     <Company>Andrew Arnott</Company>
     <Copyright>Copyright © Andrew Arnott</Copyright>
     <Authors>Andrew Arnott</Authors>
     <RepositoryUrl>https://github.com/aarnott/CodeGeneration.Roslyn</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
-    <IsPackable Condition=" $(MSBuildProjectName.Contains('Test')) ">false</IsPackable>
-    <NerdbankGitVersioningVersion>2.2.13</NerdbankGitVersioningVersion>
+    <LicenseExpression>MS-PL</LicenseExpression>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Shared versions">
     <RoslynNugetVersion>2.9.0</RoslynNugetVersion>
   </PropertyGroup>
 
+  <PropertyGroup Label="Signing" Condition=" !$(IsTestProject) ">
```

Comment:
I'm not a fan of all the labeling. It becomes too hard to predict. For example, why wouldn't signing be part of your "build configuration" group? I'd only know to put or look for them elsewhere after reviewing every labeled group. "NuGet metadata" *seems* clear, but then `Copyright` and `Company` *also* impact the assembly attributes that are generated and compiled into the assembly, so they aren't even themselves just nuget metadata.

Reply:
I removed all except one for the "Metadata". Not sure about the name, I've thought about "Author(ing)", or maybe I should just merge it into the main group.